pkg/api/agentPoolOnlyApi/v20180331/merge_test.go

```diff
@@ -163,3 +163,240 @@ func TestMerge_EnableRBAC(t *testing.T) {
+	if e != nil {
+		t.Error("expect error to be nil")
+	}
+
+	if newMC.Properties.AADProfile == nil {
+		t.Error("AADProfile should not be nil")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppSecret == "" {
+		t.Error("ServerAppSecret did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppID != "1a34-5" {
+		t.Error("ServerAppID did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppSecret != "ba34-5" {
+		t.Error("ServerAppSecret did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ClientAppID != "1234-5" {
+		t.Error("ClientAppID did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.TenantID != "c234-5" {
+		t.Error("TenantID did not have the expected value after merge")
+	}
+
+	// Nil AAD profile was passed during update but DM had AAD Profile
+	newMC = &ManagedCluster{
+		Properties: &Properties{
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: nil,
+		},
+	}
+
+	existingMC = &ManagedCluster{
+		Properties: &Properties{
+			DNSPrefix:  "something",
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: &AADProfile{
+				ClientAppID:     "1234-5",
+				ServerAppID:     "1a34-5",
+				ServerAppSecret: "ba34-5",
+				TenantID:        "c234-5",
+			},
+		},
+	}
+
+	e = newMC.Merge(existingMC)
+	if e != nil {
+		t.Error("expect error to be nil")
+	}
+
+	if newMC.Properties.AADProfile == nil {
+		t.Error("AADProfile should not be nil")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppSecret == "" {
+		t.Error("ServerAppSecret did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppID != "1a34-5" {
+		t.Error("ServerAppID did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppSecret != "ba34-5" {
+		t.Error("ServerAppSecret did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.ClientAppID != "1234-5" {
+		t.Error("ClientAppID did not have the expected value after merge")
+	}
+
+	if newMC.Properties.AADProfile.TenantID != "c234-5" {
+		t.Error("TenantID did not have the expected value after merge")
+	}
+
+	// No AAD profile set
+	newMC = &ManagedCluster{
+		Properties: &Properties{
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: nil,
+		},
+	}
+
+	existingMC = &ManagedCluster{
+		Properties: &Properties{
+			DNSPrefix:  "something",
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: nil,
+		},
+	}
+
+	e = newMC.Merge(existingMC)
+	if e != nil {
+		t.Error("expect error to be nil")
+	}
+
+	if newMC.Properties.AADProfile != nil {
+		t.Error("AADProfile should be nil")
+	}
+
+	// Empty field in AAD profile was passed during update but DM had AAD Profile
+	newMC = &ManagedCluster{
+		Properties: &Properties{
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: &AADProfile{
+				ClientAppID:     "1234-5",
+				ServerAppID:     "1a34-5",
+				ServerAppSecret: "",
```

Comment:
Just from a workflow design perspective: is this the intention? We don't want to throw a validation-type error when the new config is incomplete like this?

Reply:
PUT is used for both create and update so we support partial updates and merge.